test_inspect fails on Windows. Example:

```

FAIL: test_details_option_with_aliased_target (test.test_inspect.test_inspect.TestModuleCLI.test_details_option_with_aliased_target)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\Lib\test\test_inspect\test_inspect.py", line 6676, in test_details_option_with_aliased_target
    self.assertEqual(output_lines, expected_lines)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Lists differ: ['Tar[116 chars], 'Cached: None', 'Line: 98', ''] != ['Tar[116 chars], 'Cached: __pycache__\\frozencpython-316.pyc', 'Line: 98', '']

First differing element 3:
'Cached: None'
'Cached: __pycache__\\frozencpython-316.pyc'

  ['Target: ntpath:join (looked up as "os:path.join")',
   'Origin: frozen',
   'Source: D:\\a\\cpython\\cpython\\Lib\\ntpath.py',
-  'Cached: None',
+  'Cached: __pycache__\\frozencpython-316.pyc',
   'Line: 98',
   '']
```

--- a/Lib/test/test_inspect/test_inspect.py
+++ b/Lib/test/test_inspect/test_inspect.py
@@ -6660,9 +6660,13 @@ class TestModuleCLI(unittest.TestCase):
         # assert_python_ok() runs the subprocess in isolated mode (-I), which
         # ignores PYTHONPYCACHEPREFIX, so compute the expected cached path the
         # same way (i.e. without any pycache prefix) to stay independent of the
-        # environment the test suite is run in.
-        with support.swap_attr(sys, 'pycache_prefix', None):
-            cached = importlib.util.cache_from_source(module.__spec__.origin)
+        # environment the test suite is run in.  For frozen modules (e.g.
+        # ntpath on Windows) there is no cached path; keep it None.
+        if module.__spec__.cached is not None:
+            with support.swap_attr(sys, 'pycache_prefix', None):
+                cached = importlib.util.cache_from_source(module.__spec__.origin)
+        else:
+            cached = None
         # Full rendering check on the expected output
         expected_lines = [
             f'Target: {defining_target} (looked up as "{cli_target}")',

--- a/Lib/test/test_inspect/test_inspect.py
+++ b/Lib/test/test_inspect/test_inspect.py
@@ -6520,6 +6520,19 @@ class TestModuleCLI(unittest.TestCase):
     NO_SOURCE_ERROR = "No source code available for defining module"
     NO_SOURCE_TARGET_ERROR = "Failed to retrieve source code for given target"
 
+    @staticmethod
+    def _expected_cached(module):
+        # assert_python_ok() runs the subprocess in isolated mode (-I), which
+        # ignores PYTHONPYCACHEPREFIX, so compute the expected cached path the
+        # same way (i.e. without any pycache prefix) to stay independent of the
+        # environment the test suite is run in.  Modules without a cached path
+        # (e.g. frozen modules such as ntpath/importlib.machinery on Windows)
+        # report None, so preserve that.
+        if module.__spec__.cached is None:
+            return None
+        with support.swap_attr(sys, 'pycache_prefix', None):
+            return importlib.util.cache_from_source(module.__spec__.origin)
+
     def test_only_source(self):
         module = importlib.import_module('unittest')
         rc, out, err = assert_python_ok('-m', 'inspect',
@@ -6577,12 +6590,7 @@ class TestModuleCLI(unittest.TestCase):
         args = support.optim_args_from_interpreter_flags()
         rc, out, err = assert_python_ok(*args, '-m', 'inspect',
                                         module_name, '--details')
-        # assert_python_ok() runs the subprocess in isolated mode (-I), which
-        # ignores PYTHONPYCACHEPREFIX, so compute the expected cached path the
-        # same way (i.e. without any pycache prefix) to stay independent of the
-        # environment the test suite is run in.
-        with support.swap_attr(sys, 'pycache_prefix', None):
-            cached = importlib.util.cache_from_source(module.__spec__.origin)
+        cached = self._expected_cached(module)
         # Full rendering check on the expected output
         expected_lines = [
             f"Target: {module.__name__}",  # No aliasing
@@ -6626,12 +6634,7 @@ class TestModuleCLI(unittest.TestCase):
         args = support.optim_args_from_interpreter_flags()
         rc, out, err = assert_python_ok(*args, '-m', 'inspect',
                                         cli_target, '--details')
-        # assert_python_ok() runs the subprocess in isolated mode (-I), which
-        # ignores PYTHONPYCACHEPREFIX, so compute the expected cached path the
-        # same way (i.e. without any pycache prefix) to stay independent of the
-        # environment the test suite is run in.
-        with support.swap_attr(sys, 'pycache_prefix', None):
-            cached = importlib.util.cache_from_source(module.__spec__.origin)
+        cached = self._expected_cached(module)
         # Full rendering check on the expected output
         # The error is only informational when reading source details
         expected_lines = [
@@ -6657,16 +6660,7 @@ class TestModuleCLI(unittest.TestCase):
         args = support.optim_args_from_interpreter_flags()
         rc, out, err = assert_python_ok(*args, '-m', 'inspect',
                                         cli_target, '--details')
-        # assert_python_ok() runs the subprocess in isolated mode (-I), which
-        # ignores PYTHONPYCACHEPREFIX, so compute the expected cached path the
-        # same way (i.e. without any pycache prefix) to stay independent of the
-        # environment the test suite is run in.  For frozen modules (e.g.
-        # ntpath on Windows) there is no cached path; keep it None.
-        if module.__spec__.cached is not None:
-            with support.swap_attr(sys, 'pycache_prefix', None):
-                cached = importlib.util.cache_from_source(module.__spec__.origin)
-        else:
-            cached = None
+        cached = self._expected_cached(module)
         # Full rendering check on the expected output
         expected_lines = [
             f'Target: {defining_target} (looked up as "{cli_target}")',

--- a/Misc/NEWS.d/next/Tests/2026-06-22-19-45-00.gh-issue-151626.K9pZ2x.rst
+++ b/Misc/NEWS.d/next/Tests/2026-06-22-19-45-00.gh-issue-151626.K9pZ2x.rst
@@ -1,5 +1,5 @@
-Fix several tests in :mod:`test.test_inspect`, :mod:`test.test_import`,
-:mod:`test.test_importlib`, :mod:`test.test_py_compile` and
-:mod:`test.test_compileall` that failed when the test suite was run with
+Fix several tests in ``test.test_inspect``, ``test.test_import``,
+``test.test_importlib``, ``test.test_py_compile`` and
+``test.test_compileall`` that failed when the test suite was run with
 :envvar:`PYTHONPYCACHEPREFIX` set.  These tests now neutralize the pycache
 prefix where they assume the default ``__pycache__`` bytecode layout.

